presto-orc/src/main/java/com/facebook/presto/orc/Filters.java

```diff
@@ -714,4 +747,266 @@ public static Filter createMultiRange(List<Filter> filters, boolean nullAllowed)
+        {
+            if (failNext > 0) {
+                filterIdx++;
+                failNext--;
+                return false;
+            }
+            Filter filter = nextFilter();
+            if (filter != null) {
+                return processResult(filter.testLong(value));
+            }
+            return true;
+        }
+
+        @Override
+        public boolean testDouble(double value)
+        {
+            if (failNext > 0) {
+                filterIdx++;
+                failNext--;
+                return false;
+            }
+            Filter filter = nextFilter();
+            if (filter != null) {
+                return processResult(filter.testDouble(value));
+            }
+            return true;
+        }
+        @Override
+        public boolean testFloat(float value)
+        {
+            if (failNext > 0) {
+                filterIdx++;
+                failNext--;
+                return false;
+            }
+            Filter filter = nextFilter();
+            if (filter != null) {
+                return processResult(filter.testFloat(value));
+            }
+            return true;
+        }
+
+        @Override
+        public boolean testDecimal(long low, long high)
+        {
+            if (failNext > 0) {
+                filterIdx++;
+                failNext--;
+                return false;
+            }
+            Filter filter = nextFilter();
+            if (filter != null) {
+                return processResult(filter.testDecimal(low, high));
+            }
+            return true;
+        }
+
+        @Override
+        public boolean testBoolean(boolean value)
+        {
+            if (failNext > 0) {
+                filterIdx++;
+                failNext--;
+                return false;
+            }
+            Filter filter = nextFilter();
+            if (filter != null) {
+                return processResult(filter.testBoolean(value));
+            }
+            return true;
+        }
+
+        @Override
+        public boolean testBytes(byte[] value, int offset, int length)
+        {
+            if (failNext > 0) {
+                filterIdx++;
+                failNext--;
+                return false;
+            }
+            Filter filter = nextFilter();
+            if (filter != null) {
+                return processResult(filter.testBytes(value, offset, length));
+            }
+            return true;
+        }
+
+        @Override
+        public Filter nextFilter()
+        {
+            filterIdx++;
+            verify(filterIdx < numFilters);
+            return allFilters ? filters[filterIdx] : filters[filterOrder[filterIdx]];
+        }
+
+        private boolean processResult(boolean result)
+        {
+            if (result == false) {
+                // The remaining elements of the containing array/map will also be disqualified.
+                if (allFilters) {
+                    int nthArray = filterContainer[filterIdx];
```

Comment:
I'd rename nthArray to inputNumber and fold if (filterContainer[i] != nthArray) check into for loop like so:

```
            if (scanAllPositions) {
                int inputNumber = inputNumbers[filterIndex];
                for (int i = filterIndex + 1; i < numPositions && inputNumbers[i] == inputNumber; i++) {
                    numNextPositionsToFail++;
                }
            }
            else {
                int inputNumber = inputNumbers[selectedPositionIndexes[filterIndex]];
                for (int i = filterIndex + 1; i < numSelectedPositions && inputNumbers[selectedPositionIndexes[i]] == inputNumber; i++) {
                    numNextPositionsToFail++;
                }
            }
```